pkg/client/vault/client.go

```diff
@@ -170,16 +203,84 @@ func (c *Client) CreateVault() (*CreatedVault, error) {
 	}, nil
 }
 
+// CreateAuthorization creates a new authorization.
+// nolint: funlen
+func (c *Client) CreateAuthorization(vaultID, requestingParty string, scope *Scope) (*CreatedAuthorization, error) {
+	info, err := c.getVaultInfo(vaultID)
+	if err != nil {
+		return nil, fmt.Errorf("get vault info: %w", err)
+	}
+
+	kh, err := c.kms.Get(info.KID)
+	if err != nil {
+		return nil, fmt.Errorf("kms get: %w", err)
+	}
+
+	kmsCapability, err := uncompressZCAP(info.Auth.KMS.AuthToken)
+	if err != nil {
+		return nil, fmt.Errorf("kms uncompressZCAP: %w", err)
+	}
+
+	kmsNewCapability, err := zcapld.NewCapability(&zcapld.Signer{
+		SignatureSuite:     ed25519signature2018.New(suite.WithSigner(newSigner(c.crypto, kh))),
+		SuiteType:          ed25519signature2018.SignatureType,
+		VerificationMethod: vaultID,
+	}, zcapld.WithParent(kmsCapability.ID), zcapld.WithInvoker(requestingParty),
+		zcapld.WithAllowedActions("unwrap"),
+		zcapld.WithInvocationTarget(kmsCapability.InvocationTarget.ID, kmsCapability.InvocationTarget.Type),
```

Comment:
TODO: this is pointing to the keystore and not the specific key, should be the unwrapping key. Requires a change in hub-kms, likely a new endpoint to retrieve zcaps for more fine-grained resources.